rero_ils/modules/items/api/circulation.py

```diff
@@ -793,15 +793,34 @@ def prior_checkout_actions(self, action_params):
             ):
                 item, receive_actions = self.receive(**action_params)
                 actions.update(receive_actions)
-            if loan['state'] == LoanState.ITEM_IN_TRANSIT_TO_HOUSE:
-                item, cancel_actions = self.cancel_loan(pid=loan.get('pid'))
+            elif loan['state'] == LoanState.ITEM_IN_TRANSIT_TO_HOUSE:
+                item, cancel_actions = self.cancel_loan(**action_params)
                 actions.update(cancel_actions)
                 del action_params['pid']
         else:
             loan = get_loan_for_item(item_pid_to_object(self.pid))
             if (loan and loan['state'] != LoanState.ITEM_AT_DESK):
                 item, cancel_actions = self.cancel_loan(pid=loan.get('pid'))
                 actions.update(cancel_actions)
+        # CHECKOUT_1_2_2: checkout denied if some pending loan are linked to it
+        # with different patrons
+        # Except while coming from an ITEM_IN_TRANSIT_TO_HOUSE loan because
+        # the loan is cancelled and then came up in ON_SHELF to be checkout
+        # by the second patron.
+        if self.status == ItemStatus.ON_SHELF and \
+                loan['state'] != LoanState.ITEM_IN_TRANSIT_TO_HOUSE:
+            item_pid = item_pid_to_object(self.pid)
```

Comment:
you can replace the following lines by using the method:
`self.get_item_loans_by_state(state= LoanState.PENDING)`